hack/make-rules/build_with_container.sh

```diff
@@ -23,11 +23,12 @@ set -o pipefail
 KUBEEDGE_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd -P)"
 MOUNTPATH="${MOUNTPATH:-/kubeedge}"
 KUBEEDGE_BUILD_IMAGE=${KUBEEDGE_BUILD_IMAGE:-"kubeedge/build-tools"}
-DOCKER_GID="${DOCKER_GID:-$(grep '^docker:' /etc/group | cut -f3 -d:)}"
+USERNAME="$UID"
+DOCKER_GID="${DOCKER_GID:-$(grep '^docker:' /etc/group | cut -f3 -d:)}" && USERNAME="${UID}:${DOCKER_GID}"
```

Comment:
what's the difference before and after this fix? Isn't that exactly the same? :)

Reply:
To prevent the script from terminating with exit code 1 when it can't find the gid of docker.

Reply:
on mac, there's no file `/etc/group`, and then `grep '^docker:' /etc/group | cut -f3 -d:)` cmd will fail. And by running `$?` cmd, we will find the exit code is not 0.
And on mac, by appending `&& USERNAME="${UID}:${DOCKER_GID}"`, the command will run successfully and the exit code will be 0
That's why this patch will fix on mac terminating with exit code 1